### PR TITLE
Add a guide for USB Host asynchronous transfers

### PR DESCRIPTION
**Pull request by mbloch2**
Add a guide for USB Host asynchronous transfers.

ACR-886

### CODE REVIEW

**Comment by Reni-90** [comment]
@mbloch2,
The content added here in **Asynchoronus Transfer** is copied from the Tizen API reference document, https://docs.tizen.org/application/native/api/mobile/latest/group__CAPI__USB__HOST__ASYNC__MODULE.html

As this is Tizen content only, there won't be any copyright issues.
However, duplicating content on a website is not a good practice.

Therefore, I suggest briefing about the topic in the overview part of the guide and linking the same to the API reference link, https://docs.tizen.org/application/native/api/mobile/latest/group__CAPI__USB__HOST__ASYNC__MODULE.html
or
we can elaborate on the topic  **Asynchoronus Transfer** more in the **USB Host** guide such as additional information, points missed in the API reference document, and so on.

Please verify and share your valuable opinion.

@mijin85cho 
Request your valuable suggestion on the same.

**Comment by mijin85cho** [comment]
As @Reni-90 commented, duplicated content will increase not the usefulness but the effort to maintain the content. So please elaborate on the topic more than API page if possible.

**Comment by mijin85cho** [comment]
Or, reducing the content in the API Reference file can be one possible way, as the duplicated part is the overview of API Reference page.

**Comment by mbloch2** [comment]
I agree about duplication being undesirable, I didn't know about the copy of this content in the API reference (I took it from the source code, whence I assume the reference API also took it). Maybe the existence of this content in the API reference would be enough for the ACR in which case no change would be needed - I'll ask. If something is still needed (likely, since I was specifically directed to the file I modified) I think I'll just add a link to the API reference page instead.

**Comment by Reni-90** [comment]
@mbloch2 
Request you to please update the file as suggested in the previous comments.

File: docs/application/native/guides/connectivity/usb-host.md [hunk 1]
```diff
@@ -237,6 +237,12 @@ usb_host_release_interface(interface);
 dlog_print(DLOG_INFO, LOG_TAG, "%d bytes of response received: %s", bytes_transferred, buffer);
 ```
 
+<a name="async_transfer"></a>
+## Asynchronous Transfer
+
+Asynchronous transfers, including isochronous and control transfers, are explained with examples on [their API page](../../api/mobile/latest/group__CAPI__USB__HOST__ASYNC__MODULE.html).
```
Comment (Reni-90) [inline docs/application/native/guides/connectivity/usb-host.md]: To make it more readable and to maintain consistency with the Tizen docs, can we explain what is asynchronous transfer and link the API page. For example,

## Asynchronous Transfer

Asynchronous transfer is...
Asynchronous transfer includes isochronous and control transfers.
For more information, see [Asynchronous IO](../../api/mobile/latest/group__CAPI__USB__HOST__ASYNC__MODULE.html).

Please verify.

**Comment by mbloch2** [comment]
Modified as per review.

File: docs/application/native/guides/connectivity/usb-host.md [hunk 2]
```diff
@@ -240,8 +240,8 @@ dlog_print(DLOG_INFO, LOG_TAG, "%d bytes of response received: %s", bytes_transf
 <a name="async_transfer"></a>
 ## Asynchronous Transfer
 
-Asynchronous transfer is a way to let the system handle traffic in the background, without having to wait for a transfer to finish. A program can request an asynchronous transfer and proceed further, then be notified by the system at some point after the request gets handled.
-Asynchronous transfer includes isochronous and control transfers. For more information, including examples, see [Asynchronous IO](../../api/mobile/latest/group__CAPI__USB__HOST__ASYNC__MODULE.html).
+Asynchronous transfer is a way to allow the system to handle the data traffic in the background without having to wait for the transfer to finish. A program can request an asynchronous transfer and proceed with its operations. The program is notified by the system as soon as the request is handled.
```
Comment (Reni-90) [inline docs/application/native/guides/connectivity/usb-host.md]: Please change "the transfer" to "a transfer" as mentioned in the previous comment.
For more information, see https://github.com/Samsung/tizen-docs/pull/1004#discussion_r372739931

**Comment by mbloch2** [comment]
I figured this sentence should use "the" because the transfer that the system doesn't have to wait on is the specific asynchronous one (for example you can start an asynchronous transfer, then a regular transfer - the system won't have to wait for the asynchronous one but still has to wait for the regular one).

Still, modified to "a".

**Comment by Reni-90** [comment]
@mbloch2 
Thank you very much for the explanation. 
Then, to be more specific, can we elaborate it as:
Asynchronous transfer is a way to allow the system to handle the data traffic in the background without having to wait for the asynchronous transfer to finish. A program can request an asynchronous transfer and proceed with its operations. The program is notified by the system as soon as the request is handled.

Please verify whether the technical meaning is intact and share your opinion on the same.

**Comment by mbloch2** [comment]
Yes, that better reflects the technical meaning. Modified, thanks.

**Comment by mbloch2** [comment]
@journey2w ping ( https://code.sec.samsung.net/jira/browse/ACR-886 )

**Pull request merged**